meta-client/src/main/java/org/apache/helix/metaclient/factories/MetaClientConfig.java

```diff
@@ -0,0 +1,115 @@
+package org.apache.helix.metaclient.factories;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+class MetaClientConfig {
+
+  public enum StoreType {
+    ZOOKEEPER, ETCD
+  }
+
+  private final String _connectionAddress;
+  private final long _connectionTimeout;
+  private final boolean _enableAuth;
+  private final StoreType _storeType;
```

Comment:
ZkClients has many more variables than this (operation timeout, serializer). How do they fit in this picture?

Reply:
Right. Those will be added in following commits when adding ZkMetaClient config. 
Some variables will be added here if it is generals enough.